control/pages.go

```diff
@@ -3,23 +3,25 @@ package control
 import (
 	"fmt"
 	"github.com/Parquery/mailgun-relayery/database"
+	"github.com/Parquery/mailgun-relayery/protoed"
 	"math"
 )
 
-// channelPageResponse computes the response for a channels page listing
+// paginateChannels computes the response for a channels page listing
 // request. If the page is out of bounds, the response contains no channels.
 //
-// channelPageResponse requires:
+// paginateChannels requires:
 // * db != nil
 // * page != 0
 // * perPage != 0
 //
-// channelPageResponse ensures:
-// * channelsPage.Page == int32(page)
-// * channelsPage.PerPage == int32(perPage)
-// * channelsPage.PageCount >= 0
-func channelPageResponse(page uint, perPage uint,
-	db *database.Database) (channelsPage ChannelsPage, err error) {
+// paginateChannels ensures:
+// * err != nil || channelsPage.Page == int32(page)
+// * err != nil || channelsPage.PerPage == int32(perPage)
+// * err != nil || channelsPage.PageCount >= 0
+// * err != nil || len(channelsPage.Channels) <= int(perPage)
```

Comment:
I'd also add a contract that, if inTest and Count is 0 and no err, the page is 0, the page count is 0.

Reply:
Pages are 1-based, so there is no page 0. I'll add the condition with page == 1.